xxhsum.c

```diff
@@ -1811,19 +1813,39 @@ static CanonicalFromStringResult canonicalFromString(unsigned char* dst,
  *
  *      <8, 16, or 32 hexadecimal char> <space> <space> <filename...> <'\0'>
  */
-static ParseLineResult parseLine(ParsedLine* parsedLine, const char* line)
+static ParseLineResult parseLine(ParsedLine* parsedLine, char* line, int rev)
 {
-    const char* const firstSpace = strchr(line, ' ');
-    if (firstSpace == NULL) return ParseLine_invalidFormat;
+    char* const firstSpace = strchr(line, ' ');
+    const char* hash_ptr;
+    size_t hash_len;
+
+    if (firstSpace == NULL || !firstSpace[1]) return ParseLine_invalidFormat;
 
     parsedLine->filename = NULL;
     parsedLine->xxhBits = 0;
 
-    switch (firstSpace - line)
+    if (firstSpace[1] == '(') {
+        char* lastSpace = strrchr(line, ' ');
+        if (lastSpace - firstSpace < 5) return ParseLine_invalidFormat;
+        if (lastSpace[-1] != '=' || lastSpace[-2] != ' ' || lastSpace[-3] != ')') return ParseLine_invalidFormat;
+        lastSpace[-3] = '\0'; /* Terminate the filename */
+        *firstSpace = '\0';
```

Comment:
This action is modifying the input.
I'm wondering if there could be another way, preserving immutability of input ?

Reply:
Why? It's just the input line. The code doesn't need to keep it unchanged. You could create a static buff of the same size as the input line if you really wanted to, though.

Reply:
Let's keep it at 7032ced649eefec83ad45fa923e72d17a3e99e9a then.

Reply:
OK. I discarded the commits past that point.